R/module_nested_tabs.R

```diff
@@ -263,7 +263,7 @@ srv_teal_module.teal_module <- function(id,
       is_transform_failed <- reactiveValues()
       transformed_teal_data <- srv_transform_teal_data(
         "data_transform",
-        data = filtered_teal_data,
+        teal_data_r = filtered_teal_data,
```

Comment:
@llrs-roche since `srv_transform_teal_data` is exported, and you changed the name of the parameter, then I would expect you also provide a PR to `tmg` and `tmc` with this change. 
https://github.com/insightsengineering/teal.modules.general/blob/main/R/utils.R#L306
Please also adjust `srv_decorate_teal_data`

Reply:
I will send the accompanying PRs. I first wanted to have the renaming approved before start opening PR to other repositories.

Reply:
yes, I think we can go with `r_teal_data`

Reply:
I'm not a fan of renaming `data` to both `teal_data_r` and `r_teal_data`. In the server of teal_module we have argument `data` which is a "reactive returning teal_data object".  `srv_decorate_teal_data` have `data` argument which is exactly the same class. Module dev is going to have two different argument names for the same object in the exported functions just to "adjust" names with .

This is double sided sword, as we have two definitions of `data`: `module$server` has `data` described above and `init` has an argument `data` (teal_data (static), reactive teal_data and teal_data_module).

-----

Let's consider perspective of two different users:

- app developer provides a `teal_data` to the `teal::init(data)` and specifies `tm_xyz(transformators)`. `transformator` has a server `function(id, data)` where `data` is a `reactive teal_data`. App developer might be confused by this, assuming that transformator's `data` is the same object.
- module developer makes the code of the server where `data` is a `reactive teal_data`. This `data` object is chained via some reactives (usually prefixed something_q) and passed to the `srv_decorate_teal_data(data)`. Changing to `srv_decorate_teal_data(teal_data_r or r_teal_data)` is nto compatible with current naming standard in the module's server, where we use:
    - <something>_r - reactive returning object like plot, data.frame (often taken from inside of teal_data).
    - <something>_q - reactive returning teal_data (qenv) object.

Reply:
If we decide to name "reactive teal_data" from `data` to `r_data` then we should be consistent and change it:
- in `module(server)` ⚠️ 
- in `teal_transform_module(server)`
-  `srv_transform_teal_data`, `srv_decorate_teal_data`

Reply:
Yes, I feared that changes in internal parameters could end up leaking to major exported functions. 

It seems that to be more consistent we would need to modify exported functions. @pawelru, any preference or comment on this discussion?

Reply:
Thanks for the ping. I think that the teal framework is pretty mature hence I think I would rather prefer limiting breaking changes